### ISSUE TITLE
Duplicated tags when the key attribute is used

### ISSUE THREAD

**Issue by paulnta** [unjs/unhead#135]
### Environment

- Node v18
- Vue v3.2.47

### Reproduction

https://stackblitz.com/edit/nuxt-unhead-tags-duplication?file=app.vue

### Describe the bug

When using `useHead` with SSR, adding `key` (or `hid`, `vmid`) will cause scripts to be duplicated after hydration on the client.

```js
useHead({
  script: [
    { innerHTML: "console.log('will log once')" },
    {
      key: 'foo',
      innerHTML: "console.log('will log twice')",
    },
  ],
});
```

As stated in [the documentation](https://unhead.harlanzw.com/guide/getting-started/ssr#3-done-how-hydration-works), I would expect the tag with `key: 'foo'` to be deduplicated. Instead, it's injected twice in the page head. 
> When your client-side app hydrates the server head tags, it will attempt to hydrate each element based on the nodes being equal $el.isEqualNode($newEl) or them sharing the same dedupe key (see [Tag Deduping](https://unhead.harlanzw.com/guide/guides/handling-duplicates)).
  

### Additional context

When using `useHead` in server mode (`useHead(input, { mode: 'server' })`), the deduplication works correctly because the `data-h-xxx` attribute is generated.

**Comment by harlan-zw** [comment]
Thank you for clarifying, this is not the expected behavior. Working on a fix

**Comment by harlan-zw** [comment]
Fix is in https://github.com/unjs/unhead/pull/138, will try and get a release out tomorrow with it after I can sanity check it with fresh eyes

**Comment by harlan-zw** [comment]
Should be fixed in 1.1.25, please let me know if you have any further issues with it.
